### PR TITLE
feature: Add GET spendings query

### PR DESCRIPTION
**Pull request by anastasiya-nd**

### CODE REVIEW

**Review by ok2ju — CHANGES_REQUESTED** [review]

File: src/redux/sagas.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'regenerator-runtime/runtime';
+import { takeEvery, put, call, all } from '@redux-saga/core/effects';
+import spendings from '../api/api';
```
[the remark this answers is not in the document]
Comment (anastasiya-nd) [reply src/redux/sagas.js:3]: > Did you like path like that? `/api/api`?

Should I create index.js?

[the remark this answers is not in the document]
Comment (ok2ju) [reply src/redux/sagas.js:3]: I hope so

File: src/redux/actions.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { GET_SPENDINGS, REQUEST_SPENDINGS } from './types';
+
+export const getSpendings = (spendings, pagination) => { //eslint-disable-line
```
[the remark this answers is not in the document]
Comment (anastasiya-nd) [reply src/redux/actions.js:3]: > As I mentioned below, for actions that make network requests in most cases it's common practice to have 3 types of one action (_PENDING, _SUCCESS, _ERROR). So, the same logic applies here. There will be 3 action creators: `requestSpendingsPendings`, `requestSpendingsSuccess`, `requestSpendingsError`.

Am I thinking correctly: 
-'getSpendings' has to be 'requestSpendingsSuccess',
-'requestSpendings' has to be 'requestSpendingsPendings'
-and i have to add 'requestSpendingsError' to 'getSpendingsWorker(data)' ?

[the remark this answers is not in the document]
Comment (ok2ju) [reply src/redux/actions.js:3]: Yes, and `requestSpendingsError` will be in the `catch` block accepting `error`

File: src/redux/reducer.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  pagination: {
+    total: undefined,
+    page: 1,
+    perPage: 5,
+  },
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case REQUEST_SPENDINGS_SUCCESS:
```
Comment (ok2ju) [inline src/redux/reducer.js:14]: Why don't you use `REQUEST_SPENDINGS_PENDING` and `REQUEST_SPENDINGS_ERROR` actions here? You can add `loading` filed into `initialState` and show loading screen on your page while request is in process (`REQUEST_SPENDINGS_PENDING`) and hide it when request is finished (`REQUEST_SPENDINGS_SUCCESS`, `REQUEST_SPENDINGS_ERROR`).

File: src/redux/reducer.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      return {
+        ...state,
+        spendings: [...action.spendings],
+        pagination: { ...action.pagination },
```
Comment (ok2ju) [inline src/redux/reducer.js:18]: If `spendings` is already an array, why do you use spreading here? The same for `pagination`.

File: src/redux/sagas.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ok2ju) [reply src/redux/sagas.js:3]: You added `index` file but you didn't change import path here.

File: src/redux/sagas.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+    const response = yield call(spendings.getSpendings, data.page, data.perPage);
+    yield put(requestSpendingsSuccess(response.data.spendings, response.data.pagination));
+  } catch (error) {
+    yield put(requestSpendingsError());
+  }
+}
+
+export default function* rootSaga() {
+  yield all([takeEvery(REQUEST_SPENDINGS_PENDING, getSpendingsWorker)]);
```
Comment (ok2ju) [inline src/redux/sagas.js:17]: You should use `takeLatest` here instead of `takeEvery`.

File: src/redux/types.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+/* eslint-disable */
+export const REQUEST_SPENDINGS_PENDING ='REQUEST_SPENDINGS_PENDING',
```
Comment (ok2ju) [inline src/redux/types.js:2]: Please, define separate export for each variable.

**Review by ok2ju — CHANGES_REQUESTED** [review]

**Review by ok2ju — APPROVED** [review]

**Pull request merged**